generator/generator.py

```diff
@@ -1,31 +1,67 @@
-#import numpy
+import numpy as np
 import sys
 import csv
 import random
+import matplotlib.pyplot as plt
+import scipy.stats
 
 def generator(shards, deltaT, N, mean, std):
-    tmp_list = []
     vectors = []
+    tasks_in_deltaT = []
+    requests = []
+    time_marks_in_deltaT = []
 
-    for x in range(shards):
-        for y in range(N):
-            tmp_list.append(random.randint(0, 10))
-        vectors.append(tmp_list)
-        tmp_list = []
-
+    for z in range(shards):        
+        y = np.random.normal(mean, std, N).tolist()
+        y = [round(num, 3) for num in y]
+        vectors.append(y)
+        y = []
 
     vectors_file = open("./generator/vectors.csv","w")
-    for x in vectors:
-        vector_string = str(vectors.index(x)+1) +","+ ','.join(map(str, x)) + "\n"
-        vectors_file.write(vector_string)
-
-
+    for z in vectors:
+       vector_string = str(vectors.index(z)+1) +","+ ','.join(map(str, z)) + "\n"
+       vectors_file.write(vector_string)
+    vectors_file.close()
+            
+    for n in range(N):
+        tmp = 0
+        for v in vectors:
+            if(v[n]>0):
+                tmp = tmp + 1
+        tasks_in_deltaT.append(tmp)
 
 
+    for n in range(N):
+        tmp_deltaT_start = n * deltaT
+        tmp_deltaT_end = (n + 1) * deltaT
+        time_marks = np.random.uniform(tmp_deltaT_start, tmp_deltaT_end, tasks_in_deltaT[n])
+        time_marks = [round(tm, 2) for tm in time_marks]
+        time_marks_in_deltaT.append(time_marks)
 
+    tmp_req = []
+    for n in range(N):
+        task_iterator = tasks_in_deltaT[n] - 1
```

Comment:
we can create function that will give us next id
it could store last generated value and increment it by 1

then, sorting wouldn't be needed

Reply:
don't know what you mean, i know how to write function you mean, but i don't know what are you reffering to

Reply:
i think we always need sorting, because the output data should be sorted by timestamp.
The lists containing timestamps for each delta_t are unordered.
The way the requests are created right now (assigning shard and size to timestamp, sorting the requests and then adding to each of them id) gives us that in the output, shards appear in random order.